examples/circuit.rs

```diff
@@ -0,0 +1,95 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use dusk_plonk::prelude::*;
+use rand_core::OsRng;
+
+// Create a `TestCircuit` struct.
+#[derive(Debug, Default)]
+pub struct TestCircuit {
+    pub a: BlsScalar,
+    pub b: BlsScalar,
+    pub c: BlsScalar,
+    pub d: BlsScalar,
+    pub e: JubJubScalar,
+    pub f: JubJubAffine,
+}
+
+// Implement the 'Circuit' trait that checks:
+// 1) a < 2^6
+// 2) b < 2^4
+// 3) a + b + 42 = c where c is public input
+// 4) a * b + d = 42
+// 5) JubJub::GENERATOR * e(JubJubScalar) = f where F is a Public Input
+impl Circuit for TestCircuit {
+    fn circuit(&self, composer: &mut Composer) -> Result<(), Error> {
+        let a = composer.append_witness(self.a);
+        let b = composer.append_witness(self.b);
+        let d = composer.append_witness(self.d);
+
+        // 1) a < 2^6
+        const HALF_SIX: usize = 3;
```

Comment:
I remember that you explained me the reason for that, but I can't remember it now. As such, maybe would be a good moment to fix this? Probably more people will have the same doubt...

Reply:
The range component can only operate with even numbers. This is why we give it half of the number we want to operate on, see [here](https://github.com/dusk-network/plonk/blob/master/src/composer.rs#L1002-L1011).

I was trying to convey this by introducing the constant and naming it `HALF_*` where `*` is the value that we are checking on. As much as I can see that this is far form ideal, I'm also not sure how to make this better.
Do you have a suggestion?

Reply:
How about this:
```rust
        // 1) a < 2^6
        composer.component_range::<3>(a); // 3 BIT_PAIRS = 6 bits

        // 2) b < 2^4
        composer.component_range::<2>(b); // 2 BIT_PAIRS = 4 bits
```

Reply:
is there a way to do a compile time check on the provided value? otherwise, we could also throw an error at runtime. e.g., if the provided number is not even, throw `NotEvenError: provided constant not even.`

Reply:
btw, here:

https://github.com/dusk-network/plonk/blob/780bbf9c46a17ca0dd90543dfe2716833a4f56bb/src/composer.rs#L1015C9-L1016C18

it says that we need an even number because we iterate over 2-bit chunks. Can we simply add a 0 bit at the beginning of the bit array if the decimal number is odd?

Reply:
Actually we introduced `BIT_PAIRS` to get rid of this `debug_assert` (see the discussion in #764 ).

In any case, any change in the range or logic components is not in the scope of this PR.

If you want we can open another issue on this to have the discussion.